source/ShipJumpNavigation.cpp

```diff
@@ -0,0 +1,182 @@
+/* ShipJumpNavigation.cpp
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "ShipJumpNavigation.h"
+
+using namespace std;
+
+#include "Ship.h"
+#include "System.h"
+
+
+
+// Calibrate this ship's jump navigation information, caching its jump costs, range, and capabilities.
+void ShipJumpNavigation::Calibrate(const Ship &ship)
+{
+	Outfit attributes = ship.Attributes();
+	Outfit baseAttributes = ship.BaseAttributes();
+
+	hasHyperdrive = attributes.Get("hyperdrive");
+	hasScramDrive = attributes.Get("scram drive");
+	hasJumpDrive = attributes.Get("jump drive");
+
+	double defaultHyperFuel = hasScramDrive ? 150. : 100.;
+	double defaultJumpFuel = 200.;
```

Comment:
These should probably be stored as constants akin to how the default jump distance is stored.